netweaver/extract_installer.sls

```diff
@@ -0,0 +1,22 @@
+{%- from "netweaver/map.jinja" import netweaver with context -%}
+{% set host = grains['host'] %}
+{% for node in netweaver.nodes if node.host == host and (netweaver.sapcar_exe_file is defined and netweaver.swpm_sar_file and netweaver.swpm_extract_dir is defined) %}
+{% set instance = '{:0>2}'.format(node.instance) %}
+{% set name = '{}_{}'.format(node.sid, instance) %}
+
+
+validate_sapcar_permission_{{ host }}_{{ name }}:
```

Comment:
I don't know if we can do this. This file might be in NFS shared folder where users don't have write permissions.

Reply:
For the SAPCAR permission, there is no recommendation as such that I could find in the docs/notes. Will it not be user responsibility to provide latest version(I will update that in the docs) and have correct permissions to executable file?
We can verify if the permissions are correct(via salt module fn) before executing the methods. Or copy it to a temp location from where we can execute it with the correct rights. What do you think about this @arbulu89 @diegoakechi

Reply:
mmm, we can think about it in the future, maybe by now checking the permissions is correct.
I don't know @Simranpal , this is a trial an error thing.
Just run the test in our libvirt, azure and aws deployments and see what happens.
I cannot really tell you more.
This is how I did in many other occasions. Try and see the result, and implement according that.

Reply:
@Simranpal @arbulu89 the OS already checks the permission and if the error is readable enough I would simple delegate this to the OS and raise the execution error if it occurs to avoid extra checks (since it will be checked by the OS anyway). Documenting that the sapcar binary provided needs to have execution permissions seem enough for me.

As you have already implemented the check, I think we can keep it, but make sure to test if it will not fail in case the file has different permission, like 755 that would still be executable but AFAIK salt will try to change the permissions. Imagine that the user can point a NFS share where sapcar is mounted with read-only access, so in that situation, I think the validation would fail the execution.